### PR TITLE
More backoff for retries and add retries for return code 4

### PR DESCRIPTION
**Pull request by benank**

### CODE REVIEW

File: src/core/src/package_managers/ZypperPackageManager.py [hunk 1]
```diff
@@ -136,7 +137,7 @@ def invoke_package_manager(self, command):
 
                 if i < self.package_manager_max_retries - 1:
                     self.composite_logger.log_warning("Exception on package manager invoke. [Exception={0}] [RetryCount={1}]".format(error_msg, str(i)))
-                    time.sleep(i + 1)
+                    time.sleep(pow(2, i + 2))
```
Comment (rane-rajasi) [inline src/core/src/package_managers/ZypperPackageManager.py:140]: Why the power function?

Comment (rane-rajasi) [inline src/core/src/package_managers/ZypperPackageManager.py:140]: Function invoke_package_manager is handling a lot of things and has become too long. Think on extracting out soem part, eg: the retries could be in a different function called from invoke_package_manager

Comment (benank) [reply src/core/src/package_managers/ZypperPackageManager.py:142]: Good idea. I split it up into a few separate functions.

File: src/core/tests/library/LegacyEnvLayerExtensions.py [hunk 2]
```diff
@@ -226,6 +226,53 @@ def run_command_output(self, cmd, no_output=False, chk_err=True):
...
+                                 "Retrieving repository 'SLE-Module-Server-Applications15-SP3-Updates' metadata ....................................................[done]\n" + \
+                                 "Building repository 'SLE-Module-Server-Applications15-SP3-Updates' cache .........................................................[done]\n" + \
+                                 "Retrieving repository 'SLE-Module-Web-Scripting15-SP3-Pool' metadata .............................................................[done]\n" + \
+                                 "Building repository 'SLE-Module-Web-Scripting15-SP3-Pool' cache ..................................................................[done]\n" + \
+                                 "Retrieving repository 'SLE-Module-Web-Scripting15-SP3-Updates' metadata ..........................................................[done]\n" + \
+                                 "Building repository 'SLE-Module-Web-Scripting15-SP3-Updates' cache ...............................................................[done]\n" + \
+                                 "All repositories have been refreshed."
```
Comment (kjohn-msft) [inline src/core/tests/library/LegacyEnvLayerExtensions.py:275]: Trim it to the minimum required for the test to work.

**Review by kjohn-msft — APPROVED** [review]

File: src/core/tests/library/LegacyEnvLayerExtensions.py (hunk 2 above)
Comment (benank) [reply src/core/tests/library/LegacyEnvLayerExtensions.py]: Done

**Pull request merged**